### ISSUE TITLE
valid set overfitting?

### ISSUE THREAD

**Issue by JustM57** [catboost/catboost#191]
I've found a bug or a feature:
catClassifier get different score on the same set in cases it was used as valid or test.

**Comment by unnir** [comment]
which version are you using? 

I've noticed some over-fitting in my train results too...

**Comment by JustM57** [comment]
0.5.0

**Comment by annaveronika** [comment]
This is not a bug, this is a feature. We calculate statistics on categorical features without use of label using test set. Like frequency of a category value. This gives quality improvement and doesn't lead to overfitting, because you don't use label values.
If you want to disable it, you can set counter_calc_method to SkipTest.
